### DISCUSSION TITLE
@rearchWidget and other annotations

### DISCUSSION THREAD

**Discussion by m-sadegh-sh** [GregoryConrad/rearch-dart#208] [Q&A] [answered]
I've read the docs to see if I can utilize rearch in my next Flutter project. In the example, I can see several usages of rearch-specific annotations, e.g. rearchWidget, which tends to remove/reduce boilerplates. As I couldn't use them in a sample Flutter project, tried the beta channel, I just wanted to double-check if I missed anything regarding being able to use them.

**Comment by busslina** [comment, marked as answer]
I think they are prepared to be used when Static Metaprogramming gets ready. For now seems not ready: https://dart.dev/language/macros

**Reply by m-sadegh-sh** [reply]
So if I configure my project based on the provided steps in the link, can I assume the annotations will work as expected? (The code generation I mean)

**Reply by busslina** [reply]
Yes, it is supposed to. Last time I read the instructions there were some places where new/unavailable syntax was mixed with stable syntax.
When you see `@rearchWidget` discard it and use `RearchConsumer` instead.
https://pub.dev/documentation/flutter_rearch/latest/flutter_rearch/RearchConsumer-class.html

**Reply by GregoryConrad** [reply]
Hi @m-sadegh-sh! 👋

When/if macros land in Flutter stable, I'll release a new version of `flutter_rearch` that will have `@rearchWidget` and friends builtin. You can follow #89 for progress updates.

In the meantime, you can use `RearchConsumer` and similar; they'll still be supported after macros are released (since the macros will just generate simple boilerplate using them under the hood)
